### PR TITLE
[ADD] Param: processors (SiriusAdapter)

### PR DESCRIPTION
**Pull request by oliveralka**
Added recently to SIRIUS.
May come in handy when processing locally.

edit:
Use OpenMP check number of processors.
If OpenMP is not available - SIRIUS default -> all possible CPUs
If OpenMS is available parameter "threads" used for the number of processors.
If threads was set too high -> reset to maximal numbers of processors.

@jpfeuffer  Is it OK, to abuse the threads parameter that way?

### CODE REVIEW

**Comment by jpfeuffer** [comment]
He @oliveralka . I think you do not need to depend on OpenMP here. It should suffice to just use the threads parameter and pass it to SIRIUS.

File: src/utils/SiriusAdapter.cpp [hunk 1]
```diff
@@ -328,11 +342,15 @@ class TOPPSiriusAdapter :
...
                    << "--ppm-max" << ppm_max
                    << "--compound-timeout" << compound_timeout
-                   << "--tree-timeout" << tree_timeout 
+                   << "--tree-timeout" << tree_timeout
                    << "--quiet"
                    << "--output" << out_dir.toQString(); //internal output folder for temporary SIRIUS output file storage
 
-    // add flags
+    // check OpenMP
+    #ifdef _OPENMP
```
Comment (hroest) [inline src/utils/SiriusAdapter.cpp]: I dont understand this, this is highly surprising for users. If OpenMS was built without OpenMP support, then you cannot have multithreading in java? I dont really understand the logic of this, why not simply pass forward the `--threads` parameter?

Comment (oliveralka) [reply src/utils/SiriusAdapter.cpp]: Ok, I will do that.
So it does not matter if more threads are "called" than available - or is that not possible?

Comment (oliveralka) [reply src/utils/SiriusAdapter.cpp]: Ok, put in the case of SIRIUS you will specify the number of processors with the thread parameter.

**Comment by oliveralka** [comment]
Solution without OpenMP.
Threads are regarded as processors for SIRIUS (Number of CPUs) - is that Ok in this context?

**Comment by hroest** [comment]
>  Threads are regarded as processors for SIRIUS (Number of CPUs) - is that Ok in this context?

I think thats fine and usually the user (should) know best what s/he wants. Note that on a cluster where you have shared loads (e.g. 48 cores but you are assigned 12 of these), it is impossible to infer at runtime how many cores you are assigned and you simply need to *know* in advance and pass the correct parameter. Any changes of this could actually result in problems

**Pull request merged**